src/vs/workbench/api/test/browser/extHostTelemetry.test.ts

```diff
@@ -183,7 +183,7 @@ suite('ExtHostTelemetry', function () {
 		assert.strictEqual(functionSpy.dataArr.length, 1);
 		assert.strictEqual(functionSpy.dataArr[0].eventName, `${mockExtensionIdentifier.name}/test-event`);
 		assert.strictEqual(functionSpy.dataArr[0].data['test-data'], 'test-data');
-		assert.strictEqual(functionSpy.dataArr[0].data['common.foo'], 'bar');
+		assert.strictEqual(functionSpy.dataArr[0].data.properties['common.foo'], 'bar');
```

Comment:
I'm a bit confused why are somethings on top level data object now and some on the properties. The top level data is all the properties?

Reply:
"common.foo" is set from "additionalCommonProperties" when creating logger, it as well as other common properties will be merged into "properties" of data before calling sender method. This is why this fix can log real common properties by telemetry extesion. Other data still remain on top level.

Reply:
Hi @lramos15  do you still have question here?

Reply:
I'm still confused because some data ends up top level and others does not. We also want to avoid the properties and measurement assumption since this telemetry API is supposed to support more than just app insights and basically any arbitrary json data which is why it is done this way

Reply:
I fully understand the concern, however the properties assumption already exists regarding current logic, and there are inconsistent behaviors for different scenarios: mix common props if "properties" in data, otherwise put all common props at top level. This will affect all function callers depend on this to have condition handling.

Let's forget App Insights or other logging service now. Do you agree we should put all common props including additional common props to **one place**, no matter they are at the top level or under a reserved field? And current logic tries to put them under a field named "properties" I think. Or do you think we should put them all at top level?

The assumption of "measurements" is from the comment that no clean up for the field "measurements". Regarding to your comment, I will revert the related lines. Then I think all data record should be cleaned since no type check and no convention for "measurements".

Reply:
@lramos15  Any more comment?